operator/v1/types.go

```diff
@@ -0,0 +1,186 @@
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+}
+
+// JobConditionType defines all kinds of types of JobStatus.
+type JobConditionType string
+
+const (
+	// JobCreated means the job has been accepted by the system,
+	// but one or more of the pods/services has not been started.
+	// This includes time before pods being scheduled and launched.
+	JobCreated JobConditionType = "Created"
+
+	// JobRunning means all sub-resources (e.g. services/pods) of this job
+	// have been successfully scheduled and launched.
+	// The training is running without error.
+	JobRunning JobConditionType = "Running"
+
+	// JobRestarting means one or more sub-resources (e.g. services/pods) of this job
+	// reached phase failed but maybe restarted according to it's restart policy
+	// which specified by user in v1.PodTemplateSpec.
+	// The training is freezing/pending.
+	JobRestarting JobConditionType = "Restarting"
+
+	// JobSucceeded means all sub-resources (e.g. services/pods) of this job
+	// reached phase have terminated in success.
+	// The training is complete without error.
+	JobSucceeded JobConditionType = "Succeeded"
+
+	// JobFailed means one or more sub-resources (e.g. services/pods) of this job
+	// reached phase failed with no restarting.
+	// The training has failed its execution.
+	JobFailed JobConditionType = "Failed"
+)
+
+// CleanPodPolicy describes how to deal with pods when the job is finished.
+type CleanPodPolicy string
+
+const (
+	CleanPodPolicyUndefined CleanPodPolicy = ""
+	CleanPodPolicyAll       CleanPodPolicy = "All"
+	CleanPodPolicyRunning   CleanPodPolicy = "Running"
+	CleanPodPolicyNone      CleanPodPolicy = "None"
+)
+
+// RestartPolicy describes how the replicas should be restarted.
+// Only one of the following restart policies may be specified.
+// If none of the following policies is specified, the default one
+// is RestartPolicyAlways.
+type RestartPolicy string
+
+const (
+	RestartPolicyAlways    RestartPolicy = "Always"
+	RestartPolicyOnFailure RestartPolicy = "OnFailure"
+	RestartPolicyNever     RestartPolicy = "Never"
+
+	// `ExitCode` policy means that user should add exit code by themselves,
+	// The job operator will check these exit codes to
+	// determine the behavior when an error occurs:
+	// - 1-127: permanent error, do not restart.
+	// - 128-255: retryable error, will restart the pod.
+	RestartPolicyExitCode RestartPolicy = "ExitCode"
+)
+
+// RunPolicy encapsulates various runtime policies of the distributed training
+// job, for example how to clean up resources and how long the job can stay
+// active.
+type RunPolicy struct {
+	// CleanPodPolicy defines the policy to kill pods after TFJob is
+	// succeeded.
+	// Default to Running.
+	CleanPodPolicy *CleanPodPolicy `json:"cleanPodPolicy,omitempty"`
+
+	// TTLSecondsAfterFinished is the TTL to clean up jobs (temporary
+	// before kubernetes adds the cleanup controller).
```

Comment:
One nit, we can probably remove this comments now: `temporary before kubernetes adds the cleanup controller` now. 
since the kubernetes has already added [this](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/)

Reply:
Fixed.